stakepooldclient/stakepooldclient.go

```diff
@@ -704,3 +704,23 @@ func (s *StakepooldManager) GetStakeInfo() (*pb.GetStakeInfoResponse, error) {
 	}
 	return nil, errors.New("GetStakeInfo RPC failed on all stakepoold instances")
 }
+
+// ColdWalletExtPubMatches calls GetColdWalletExtPub RPC on all stakepoold
+// instances and compares the returned `coldwalletextpub` value against the
+// value set in dcrstakepool's config.
+// Returns false if an RPC call to any of the backend clients errors or
+// if any returned `coldwalletextpub` value is not the same as dcrstakepool's.
+func (s *StakepooldManager) ColdWalletExtPubMatches(dcrstakepoolColdWalletExtPub string) bool {
```

Comment:
I think this func should return `(bool, error)` and the calling code should have distinct logging for cases of RPC failure, and cases of incorrect coldwalletextpub config.

Right now if the RPC call fails (eg. no network) then we will log both messages:

`GetColdWalletExtPub RPC failed on stakepoold instance %s: %v`
  followed by
`coldwalletextpub is not the same in dcrstakepool.conf and stakepoold.conf files"`


We should also include the `ip:port` of the incorrectly configured stakepoold in the log message

Reply:
Is the latest implementation better?

Reply:
Looks good, thanks.